docs/cypress.config.ts

```diff
@@ -0,0 +1,38 @@
+import { defineConfig } from 'cypress';
+import sitemapUrls from 'sitemap-urls';
+import { BASE_URL } from './cypress/data/constants';
+
+const fs = require('fs');
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      // implement node event listeners here
+      on('task', {
+        readSitemapLinks: async () => {
+          if (fs.existsSync('./public/sitemap.xml')) {
+            const siteMapContent = fs.readFileSync(
+              './public/sitemap.xml',
+              'utf8'
+            );
+            const sitemapLinks: string[] = await sitemapUrls.extractUrls(
+              siteMapContent
+            );
+            return sitemapLinks.map((link) =>
+              link
+                .replace(`${BASE_URL}/`, '')
+                .replace('https://www.dev.ui.docs.amplify.aws/', '')
+                .replace('https://ui.docs.amplify.aws/', '')
+            );
+          }
+        },
+        log: (message) => {
+          console.log(message);
+          return null;
+        },
+      });
+    },
+    baseUrl: BASE_URL,
+    numTestsKeptInMemory: 0,
```

Comment:
default is `50`, reduce it to save some memory. [docs](https://docs.cypress.io/guides/references/configuration#Global)

Reply:
`The number of tests for which snapshots and command data are kept in memory` - does this mean that if we set it to 0 we won't have snapshots of failed tests? If so can we reduce to 5 or 10 instead?

Reply:
Right. We don't have the snapshot, but we still have the video...but might be good to keep the snapshot there. Let me try if we can change it to `1`. The memory issue makes it very flacky